### PR TITLE
i18n Implementation + es-MX configuration

### PR DESCRIPTION
**Pull request by Jossdz**
# TS Error translator i18n introduction
## Objective: 
Make the TS Translator available for people around the world no matter the language they speak, so they can get the best from this site and learn or improve their TS level.

## Checklist:
According to comments from the community and Matt, i18n introduction must meet these requirements:

- Use what next has built-in to get the app translated to other languages (No dependencies). ✅
- Make the app ready for new languages. ✅
	- Introduction of new languages should not require the same code (DRY). ✅
	- Translations for the errors must be in its own folder (e.g. `~/packages/engine/errors/pt-BR`). ✅
	- General translations must be on a JSON file. ✅
- Create a document to guide new collaborators through new language addition. ✅
- Introduce the new functionality without breaking the existing code (web and VS Code extension). ✅
- Translations must keep programming concepts and types in plain English. e.g.: 

```js
"A conversão do tipo '{0}' para o tipo '{1}' pode ser um erro porque nenhum tipo se sobrepõe suficientemente ao outro. Se isso foi intencional, converta a expressão para 'unknown' primeiro."
✅
```

```js
"A conversão do tipo '{0}' para o tipo '{1}' pode ser um erro porque nenhum tipo se sobrepõe suficientemente ao outro. Se isso foi intencional, converta a expressão para 'desconhecido' primeiro."
👎
```

## Changes: 

- `next.config.js` file modified to support multiple languages as sub-path routing.
- `LangSelector` component created to handle the locale change. 
- `@headlessui` and `@heroicons` installed for the `LangSelector` component.
- `apps/web/src/constants/languages.ts` created to store the available locales/languages.
- `[locale].ts` files created to handle the current UI translations.
- Translations translated to another language stored on a folder with the language id. E.g.: `packages/engine/errors/es-MX/1002.md`
- `getTranslation` function added at `apps/web/src/utils/translations.ts` to avoid import all locales when ui translations are needed.  This also makes possible add a new language easily by just add the language and the translations without touching the files reading those translations.

### CODE REVIEW

File: apps/web/src/pages/index.tsx [hunk 1]
```diff
@@ -164,30 +174,47 @@ const Query = z.object({
...
   const query = Query.parse(ctx.query);
+  const locale = ctx.locale;
+
+  const t = getTranslations(locale);
+  const isEnglish = locale === 'en' || locale === 'en-US';
+
+  console.log(
+    locale,
+    `../../packages/engine/errors/${isEnglish ? '' : locale}`,
+  );
```
Comment (mattpocock) [inline apps/web/src/pages/index.tsx:185]:
```suggestion
```

File: apps/web/src/pages/index.tsx [hunk 2]
```diff
@@ -164,30 +174,47 @@ const Query = z.object({
...
+          .map(
+            (error: any): ErrorInfo => {
+              console.log(error);
+
+              return {
+                ...error,
+                improvedError: getImprovedMessageFromMarkdown(
+                  path.resolve(
+                    process.cwd(),
+                    `../../packages/engine/errors/${isEnglish ? '' : locale}`,
```
Comment (mattpocock) [inline apps/web/src/pages/index.tsx:202]: We should move existing errors into an `en` folder

File: apps/web/src/pages/index.tsx [hunk 3]
```diff
@@ -35,7 +43,6 @@ export default function Web(props: { error: string; errors: ErrorInfo[] }) {
             firstExcerpt || `Translate TypeScript Errors to plain English`
```
Comment (mattpocock) [inline apps/web/src/pages/index.tsx:43]: Missed translation here

**Comment by mattpocock** [comment]
This proposal should also:

- Change `CONTRIBUTING.md` to include information about translations
- Handle bundling the errors for VSCode and inferring the locale there (possibly as a separate PR)

File: apps/web/src/pages/index.tsx (hunk 2 above)
Comment (Jossdz) [reply apps/web/src/pages/index.tsx:202]: This doesn't break the code for the vs code extension?

Comment (mattpocock) [reply apps/web/src/pages/index.tsx:202]: No, as long as we also _change_ the code for the vscode extension.

File: apps/web/src/pages/index.tsx (hunk 1 above)
Comment (Jossdz) [reply apps/web/src/pages/index.tsx]: Fixed on https://github.com/mattpocock/ts-error-translator/pull/92/commits/47c7bb29d032cd080221016d930ae40c62e053a6

File: apps/web/src/pages/index.tsx (hunk 2 above)
Comment (Jossdz) [reply apps/web/src/pages/index.tsx]: Fixed on https://github.com/mattpocock/ts-error-translator/pull/92/commits/47c7bb29d032cd080221016d930ae40c62e053a6

File: apps/web/src/pages/index.tsx (hunk 3 above)
Comment (Jossdz) [reply apps/web/src/pages/index.tsx]: Fixed on https://github.com/mattpocock/ts-error-translator/pull/92/commits/47c7bb29d032cd080221016d930ae40c62e053a6

**Comment by Sebkasanzew** [comment]
Will this PR also make the VSCode extension usable, if the display language of VSCode is set to another language?

For example, if VSCode display language is set to german, no extra explanations are currently shown.
Would be great to at least show the english explanations as a fallback I guess.

**Comment by PieNam** [comment]
Hello, how's it going about this pr?
Willing to add some translation after that.
